Scripts/Runtime/Entities/MemsetBufferJob.cs

```diff
@@ -0,0 +1,34 @@
+using Unity.Entities;
+using Unity.Jobs;
+using Unity.Burst;
+using Unity.Collections;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Sets all elements of a <see cref="DynamicBuffer{T}"/> to a given value.
+    /// </summary>
+    /// <typeparam name="T">The element type of the <see cref="DynamicBuffer{T}"/></typeparam>
+    /// <remarks>The <see cref="DynamicBuffer{T}"/> compliement to <see cref="MemsetNativeArray{T}"/>.</remarks>
+    [BurstCompile]
+    public struct MemsetBufferJob<T> : IJobParallelForBatch where T : struct, IBufferElementData
+    {
+        /// <summary>
+        /// The buffer to write to.
+        /// </summary>
+        [WriteOnly] public BufferFromSingleEntity<T> Source;
+        /// <summary>
+        /// The value to write to the buffer.
+        /// </summary>
+        [ReadOnly] public T Value;
+
+        public void Execute(int startIndex, int count)
+        {
+            NativeArray<T> buffer = Source.GetBuffer().AsNativeArray();
+            for (int i = startIndex; i < startIndex + count; i++)
```

Comment:
startIndex + count will get evaluated each iteration.

https://stackoverflow.com/questions/655373/is-the-condition-in-a-for-loop-evaluated-each-iteration

Might as well do a `int len = startIndex + count` and then replace the condition with len

Reply:
Huh

The linked example doesn't really address what's going on here but this link reveals that with at least the .NET compiler the value won't get cached because it's not guaranteed to be constant.
https://stackoverflow.com/questions/655373/is-the-condition-in-a-for-loop-evaluated-each-iteration

I thought it could infer that they values would be constant.

Also verified that the add operation remains at the IL level. Not sure if Burst optimizes, I couldn't quite figure out where the check was at the assembly or LLVM level.